lib/components/segmented-control/segment.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { RefObject, forwardRef, useEffect } from "react";
+import clsx from "clsx";
+import {
+    LabelPairedPlaceholderLgRegularIcon,
+    LabelPairedPlaceholderMdRegularIcon,
+    LabelPairedPlaceholderSmRegularIcon,
+} from "@deriv/quill-icons";
+import { Text } from "../Typography";
+import { SegmentedControlProps } from "./base";
+
+interface SegmentProps {
+    allowFocus?: boolean;
+    className?: string;
+    icon?: string | React.ReactNode;
+    isAnimated?: boolean;
+    isDisabled?: boolean;
+    isSelected?: boolean;
+    label?: string;
+    onClick: () => void;
+    onKeyDown: (event: React.KeyboardEvent<HTMLButtonElement>) => void;
+    size?: SegmentedControlProps["size"];
+}
+
+const placeholder = {
+    sm: <LabelPairedPlaceholderSmRegularIcon />,
+    md: <LabelPairedPlaceholderMdRegularIcon />,
+    lg: <LabelPairedPlaceholderLgRegularIcon />,
+};
+
+export const Segment = forwardRef<HTMLButtonElement, SegmentProps>(
+    (
+        {
+            allowFocus,
+            className,
+            icon,
+            isAnimated,
+            isDisabled,
+            isSelected,
+            label,
+            onClick,
+            onKeyDown,
+            size,
+        },
+        ref,
+    ) => {
+        const [focused, setFocused] = React.useState(false);
+        const Icon =
+            icon === "placeholder"
+                ? placeholder[size as keyof typeof placeholder]
+                : icon;
```

Comment:
I don't see any use case having placeholder icons here.
Can we pass placeholder icons from storybook instead? This way we can reduce the bundle size of this component. :)

Reply:
@balakrishna-deriv I had to add support for 'placeholder' because of a known issue with storybook documentation https://github.com/storybookjs/storybook/issues/17098, which gets stuck in an infinite loop when you try passing an icon as a component from within a story.